Fix broken axum-server build by pinning hyper

Currently all available versions of kitsune2_bootstrap_srv are failing to build with the following errors:
```
   Compiling axum-server v0.7.2
error[E0277]: the trait bound `<... as SendService<...>>::BodyData: Buf` is not satisfied
   --> /home/lucksus/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/axum-server-0.7.2/src/server.rs:202:87
    |
202 |                         let serve_future = builder.serve_connection_with_upgrades(io, service);
    |                                                    ------------------------------     ^^^^^^^ the trait `Buf` is not implemented for `<... as SendService<...>>::BodyData`
    |                                                    |
    |                                                    required by a bound introduced by this call
    |
    = note: required for `UpgradedSendStreamTask<<... as SendService<...>>::BodyData>` to implement `Future`
    = note: required for `TokioExecutor` to implement `Executor<hyper::proto::h2::upgrade::UpgradedSendStreamTask<<<A as accept::Accept<tokio::net::TcpStream, <M as MakeService<std::net::SocketAddr, http::Request<hyper::body::Incoming>>>::Service>>::Service as SendService<http::Request<hyper::body::Incoming>>>::BodyData>>`
    = note: required for `TokioExecutor` to implement `bounds::h2_common::Http2UpgradedExec<<<A as accept::Accept<tokio::net::TcpStream, <M as MakeService<std::net::SocketAddr, http::Request<hyper::body::Incoming>>>::Service>>::Service as SendService<http::Request<hyper::body::Incoming>>>::BodyData>`
    = note: required for `TokioExecutor` to implement `Http2ServerConnExec<TowerToHyperServiceFuture<<<A as accept::Accept<tokio::net::TcpStream, <M as MakeService<std::net::SocketAddr, http::Request<hyper::body::Incoming>>>::Service>>::Service as SendService<http::Request<hyper::body::Incoming>>>::Service, http::Request<hyper::body::Incoming>>, <<A as accept::Accept<tokio::net::TcpStream, <M as MakeService<std::net::SocketAddr, http::Request<hyper::body::Incoming>>>::Service>>::Service as SendService<http::Request<hyper::body::Incoming>>>::Body>`
    = note: required for `TokioExecutor` to implement `HttpServerConnExec<TowerToHyperServiceFuture<<<A as accept::Accept<tokio::net::TcpStream, <M as MakeService<std::net::SocketAddr, http::Request<hyper::body::Incoming>>>::Service>>::Service as SendService<http::Request<hyper::body::Incoming>>>::Service, http::Request<hyper::body::Incoming>>, <<A as accept::Accept<tokio::net::TcpStream, <M as MakeService<std::net::SocketAddr, http::Request<hyper::body::Incoming>>>::Service>>::Service as SendService<http::Request<hyper::body::Incoming>>>::Body>`
note: required by a bound in `hyper_util::server::conn::auto::Builder::<E>::serve_connection_with_upgrades`
   --> /home/lucksus/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/hyper-util-0.1.17/src/server/conn/auto/mod.rs:269:12
    |
257 |     pub fn serve_connection_with_upgrades<I, S, B>(
    |            ------------------------------ required by a bound in this associated function
...
269 |         E: HttpServerConnExec<S::Future, B>,
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Builder::<E>::serve_connection_with_upgrades`
    = note: the full name for the type has been written to '/tmp/cargo-installe88Ham/release/deps/axum_server-73df5dda139344e7.long-type-16189467587962234217.txt'
    = note: consider using `--verbose` to print the full type name to the console
help: consider further restricting the associated type
    |
159 |         A::Future: Send, <<A as accept::Accept<tokio::net::TcpStream, <M as MakeService<std::net::SocketAddr, http::Request<hyper::body::Incoming>>>::Service>>::Service as SendService<http::Request<hyper::body::Incoming>>>::BodyData: Buf
    |                          ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
```

Axum community knows about it and has a fix underway, and here they say pinning hyper to 1.7.0 helps.
https://github.com/programatik29/axum-server/issues/170

You will either need to apply the pin or update axum-server when they have a new release.

--- a/rust/sbd-server/Cargo.toml
+++ b/rust/sbd-server/Cargo.toml
@@ -20,6 +20,9 @@ axum = { workspace = true, default-features = false, features = [
   "ws",
 ] }
 axum-server = { workspace = true, features = [ "tls-rustls" ] }
+# Pin hyper to workaround issue where axum-server fails to compile.
+# TODO Remove this when upstream issue is resolved: https://github.com/programatik29/axum-server/issues/170
+hyper = "=1.7.0"
 base64 = { workspace = true }
 bytes = { workspace = true }
 clap = { workspace = true, features = [ "color", "derive", "wrap_help" ] }

